test-support/src/main/java/org/springframework/ldap/test/unboundid/EmbeddedLdapServer.java

```diff
@@ -45,27 +53,30 @@ public EmbeddedLdapServer(InMemoryDirectoryServer directoryServer) {
 		this.directoryServer = directoryServer;
 	}
 
+	/**
+	 * Creates a new {@link Builder} with a given partition suffix.
+	 *
+	 * @since 3.3
+	 */
+	public static Builder withPartitionSuffix(String partitionSuffix) {
+		return new Builder(partitionSuffix);
+	}
+
 	/**
 	 * Creates and starts new embedded LDAP server.
+	 * @deprecated Use the builder pattern exposed via
+	 * {@link #withPartitionSuffix(String)} instead.
 	 */
+	@Deprecated(since = "3.3")
 	public static EmbeddedLdapServer newEmbeddedServer(String defaultPartitionName, String defaultPartitionSuffix,
-			int port) throws Exception {
```

Comment:
Sorry that I didn't catch this earlier. We can't remove checked exceptions from signatures as this can break some forms of compatibility. Please see https://wiki.eclipse.org/Evolving_Java-based_APIs_2 and look specifically for "Delete checked exceptions thrown".